.github/workflows/destroy_review_app_infra.yml

```diff
@@ -0,0 +1,49 @@
+name: "Destroy Review App infra"
+
+on:
+  workflow_dispatch:
+    inputs:
+      key:
+        required: true
+        type: string
+
+concurrency:
+  group: review-app-infra-${{ inputs.key }}
+  cancel-in-progress: false
+
+env:
+  app_repo_role: arn:aws:iam::815624722760:role/core-infrastructure-repo
+  aws_region: eu-west-2
+
+jobs:
+  teardown:
+    name: Destroy infra
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up Terraform
+        uses: hashicorp/setup-terraform@v2
+        with:
+          terraform_version: 1.5.2
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v3
+        with:
+          aws-region: ${{ env.aws_region }}
+          role-to-assume: ${{ env.app_repo_role }}
+
+      - name: Terraform destroy
+        working-directory: terraform/development/per_review_app
+        env:
+          workspace: ${{ inputs.key }}
+        run: |
+          terraform init
+          terraform workspace select -or-create $workspace
```

Comment:
This is still `-or-create` in case it ends up running twice for any reason - would be nice to not break if the workspace has already been deleted.